### ISSUE TITLE
VS Code doesn't connect to nREPL - Part 2

### ISSUE THREAD

**Issue by eatobin** [BetterThanTomorrow/calva#40]
I have Calva working fine on my Linux machine but when I try the same setup on my Mac VS Code Calva does not connect to the running nREPL. The Calva commands can't seem to be found. Please see screenshot:

![screen shot 2018-05-14 at 8 42 53 pm](https://user-images.githubusercontent.com/870698/40032242-ea77b1c4-57b8-11e8-9bc9-97edaa4fe9fa.png)

I've attached  my `~/.lein/profiles.clj` file (which works fine on Linux).

Any idea what's wrong?

Thank you for your help,

Eric

[profiles.txt](https://github.com/BetterThanTomorrow/calva/files/2003194/profiles.txt)

**Comment by PEZ** [comment]
Hello,

Sorry for late response. Missed that there was a new issue.

I have experienced this lately as well, on Both Mac and Linux. Seems to be something with the latest Code. Not sure but there might be a pattern to it, when code starts up with showing release notes or some other virtual document.

For me it works restarting code with a Clojure file opened. Let me know if that works for you too.

We will investigate what it is (not) going on.

**Comment by eatobin** [comment]
Hello Peter,

Thanks for your reply. (I thought it was pretty quick!)

I will try your suggestion later today and report back. I also have one
more Linux machine to install and will tell you how that goes.

Thank you,

Eric

**Comment by eatobin** [comment]
Hello Peter,

Just finished my second Linux install and it is working fine:

![screenshot_2018-05-15_18-57-23](https://user-images.githubusercontent.com/870698/40089673-9453ba58-5872-11e8-9687-d1070166feb6.png)

Unfortunately, I've tried several times to get my Mac install working. No luck - still broken.

Looking forward to your research.

Thank you,

Eric

**Comment by eatobin** [comment]
Hi @pedrorgirardi,

Thanks. I tried your suggestion but still not able to make an nREPL connection on  my Mac:

![screen shot 2018-05-15 at 9 25 20 pm](https://user-images.githubusercontent.com/870698/40093295-1dc74ad4-5887-11e8-8392-f0089f16fe5d.png)

Maybe it's something besides the REPL - the Calva commands are not being found (according to the error)?

Eric

**Comment by PEZ** [comment]
That ”command not found” means the extension does not activate. Which is very strange. Are you using the same versions of Code and Calva on the Linux machine and on Mac?

**Comment by PEZ** [comment]
Hello again, I have published a new version of Calva that has an explicit Activate command: `ctrl+alt+v ctrl+alt+v`. Can you try that one out and see if it helps with this issue or not, @eatobin?

**Comment by eatobin** [comment]
Yes, all of my machines are using the same versions of Code and Calva. I
will try your newest version in a little while and let you know. E.

**Comment by PEZ** [comment]
It is a bit strange that the unicode character doesn't work on your Linux box. I'll check what happens on my machine.

**Comment by eatobin** [comment]
I think this may be a problem with my Linux distro, Manjaro (based on Arch Linux)...

On Mac:

![screen shot 2018-05-17 at 7 46 20 am](https://user-images.githubusercontent.com/870698/40201982-45d93eb4-59e6-11e8-82e5-76432cacb8e2.png)

On Manjaro:

![screenshot_2018-05-17_15-15-59](https://user-images.githubusercontent.com/870698/40202036-69fc547a-59e6-11e8-8c26-a514121254c6.png)

Am using Arial in both places in Chrome as Standard Font. Could not determine the font used in the VS Code Output Panel.

Thanks,

Eric

**Comment by eatobin** [comment]
PS - the "problems found" monkey looks fine!

![screenshot_2018-05-17_16-49-52](https://user-images.githubusercontent.com/870698/40205637-64943294-59f2-11e8-83f4-bc94e8c75682.png)

**Comment by cstettner** [comment]
Hello,

I have the same problem as it is described in this issue.
I'm working on a Mac using the latest VS Code and Calva versions. I also couldn't activate Calva with the new Activate command, because the command was not found.

Restarting VS Code like @PEZ proposed, solved the problem.

If anyone needs some more information to investigate, let me know.

**Comment by robert-johansson** [comment]
Hi @PEZ and others,

I haven't experienced any of these issues with previous versions of Calva/VS Code. However, it happens now. Maybe there was some change after the recent Calva update?

Running `code .` in a Clojure project folder leads to the problem described in this thread. However, running `code project.clj` connects to the REPL, or, enables the `calva.connect` command to work.

Curious on any solutions to this.

**Comment by robert-johansson** [comment]
An even better workaround might be to start VS Code with `code project.clj .`, with the extra dot. That will also open the sidebar and not only the clj file.

**Comment by PEZ** [comment]
I think I might have fixed the problem with Calva not being activated now. Please let me know if it shows up for you.

**Comment by robert-johansson** [comment]
My Calva is 1.3.51. That's the latest release, right?

Unfortunately the error is still there. (command 'calva.connect' not found)

**Comment by PEZ** [comment]
Grrrrr

**Comment by PEZ** [comment]
@robert-johansson It was the latest release until a few seconds ago. Now I have made a change that explicitly only activates Calva when a Clojure-ish file is open. And changed the ”Getting Started” instructions in the README accordingly. I would like to let users connect Calva to the REPL without requiring a Clojure file to be open, but I just fail in all attempts to support that, so for now this. I hope it will make it a lot easier to get started with Calva.

**Comment by gregsugiyama** [comment]
I don't know if this has been solved. I'm still getting the error "command 'calva.connect' not found.

**Comment by PEZ** [comment]
@gregsugiyama Even  if you first open a clojure file?

**Comment by PEZ** [comment]
I am sorry about this eternal issue, people. It is really a sign that the activation code is too messy.

The latest version of Calva is yet another attempt to fix the issue. All tests I can come up with tells me it is gone. But as I have thought so before many times, I again ask for your help in reporting the absence or presence of this bug back to me via this ticket. Thanks in advance!

**Comment by gregsugiyama** [comment]
@PEZ Yeah, the problem still persists. Even after upgrading to the latest version. Here are the steps:

1. Open vscode.
2. Start a lein repl
3. Open a .clj file 
4. run 'calva connect' 
5. presented with "command 'calva.connect' not found.  

I've also tried this same method while opening a .clj file first, I still get the same error

**Comment by PEZ** [comment]
@gregsugiyama Can you open the developer tools window (from the Help menu) when this error has happened and copy paste the contents here? I can't reproduce the problem with latest Calva (1.3.54) so it is very hard to try fix it. But if I can get at those error messages maybe I get some clues.

**Comment by gregsugiyama** [comment]
@PEZ Developer tools solved the problem. Turns out, I had clptn-Paredit v0.1.1 _AND_ Calva Paredit v0.1.21 installed. Something about clptn-Paredit blocks Calva from running. I deleted clptn-Paredit and now, Calva works like a charm. 

Thanks for your help!
